src/component/modal/AddressModal.jsx

```diff
@@ -0,0 +1,125 @@
+import "./addressModal.css";
+import React, { useState } from "react";
+import { useAddress } from "context/address-context";
+import { useAuth } from "context/auth-context";
+
+export const AddressModal = () => {
+  const { isAddModalOpen, setAddModalOpen, setNewAddress } = useAddress();
+  const initState = {
+    firstname: "",
+    lastname: "",
+    street: "",
+    city: "",
+    state: "",
+    pincode: "",
+    phone: "",
+    country: "India",
+  };
+  const [address, setAddress] = useState(initState);
+  const [errormsg, setErrorMsg] = useState("");
+  const changeHandler = (e) => {
+    setAddress((prev) => ({ ...prev, [e.target.name]: e.target.value }));
+  };
+  const clickHandler = (e) => {
+    e.preventDefault();
+    const { firstname, lastname, street, city, state, pincode, phone } =
+      address;
+    if (
+      firstname === "" ||
+      lastname === "" ||
+      street === "" ||
+      city === "" ||
+      state === "" ||
+      pincode === "" ||
+      phone === ""
+    ) {
+      setErrorMsg("Please enter all the fields");
+      setTimeout(()=>{
+        setErrorMsg("")
+      },1000)
+      return;
+    }
+    setErrorMsg("")
+    setNewAddress(address);
+    setAddress(initState);
+    setAddModalOpen(false);
+  };
+  return (
+    <div
+      style={isAddModalOpen ? { display: "block" } : { display: "none" }}
```

Comment:
Suggestion - Instead of inline style , it would be good to use class

Reply:
Just did for displaying modal if modal is open